dotCMS/src/integration-test/java/com/dotcms/rest/api/v1/authentication/ApiTokenResourceTest.java

```diff
@@ -0,0 +1,114 @@
+        final ApiTokenResource apiTokenResource = new ApiTokenResource(APILocator.getApiTokenAPI(), webResource);
+
+        when(request.getAttribute(WebKeys.USER)).thenReturn(limitedUser);
+        final InitDataObject initDataObject = new InitDataObject();
+        initDataObject.setUser(limitedUser);
+        when(webResource.init(any(WebResource.InitBuilder.class))).thenReturn(initDataObject);
+        final Response restResponse = apiTokenResource.revokeUserToken(request, response, limitedUser.getUserId());
+        Assert.assertNotNull(restResponse);
+        RestUtilTest.verifyErrorResponse(restResponse,  Response.Status.UNAUTHORIZED.getStatusCode(), "unauthorized to remove the token");
+    }
+
+    @Test
+    public void test_revokeUserToken() throws DotSecurityException, DotDataException, SystemException, NoSuchCompanyException {
+
+        // 1) create an user with skinid
+        // 2) call the revoke to reset the user
+        // 3) check the user has the skinid reset
+
+        final Company company = new CompanyDataGen()
+                .name("TestCompany")
+                .shortName("TC")
+                .authType("email")
+                .autoLogin(true)
+                .emailAddress("lol@dotCMS.com")
+                .homeURL("localhost")
+                .city("NYC")
+                .mx("MX")
+                .type("test")
+                .phone("5552368")
+                .portalURL("/portalURL")
+                .nextPersisted();
+        assertNotNull(company.getCompanyId());
+        final Company retrievedCompany =  CompanyUtil.findByPrimaryKey(company.getCompanyId());
+        assertEquals(company.getCompanyId(), retrievedCompany.getCompanyId());
+
+        final String skinId    = UUIDGenerator.generateUuid();
+        final User limitedUser = new UserDataGen().active(true)
+                .skinId(skinId).companyId(retrievedCompany.getCompanyId()).nextPersisted();
+
+        final User adminUser = new UserDataGen().nextPersisted();
+        APILocator.getRoleAPI().addRoleToUser(APILocator.getRoleAPI().loadCMSAdminRole(), adminUser);
+        assertTrue(APILocator.getUserAPI().isCMSAdmin(adminUser));
+
+        final HttpServletRequest  request  = mock(HttpServletRequest.class);
+        final HttpServletResponse response = mock(HttpServletResponse.class);
+        final WebResource      webResource = mock(WebResource.class);
+        final ApiTokenResource apiTokenResource = new ApiTokenResource(APILocator.getApiTokenAPI(), webResource);
+
+        when(request.getAttribute(WebKeys.USER)).thenReturn(adminUser);
+        final InitDataObject initDataObject = new InitDataObject();
+        initDataObject.setUser(adminUser);
+        when(webResource.init(any(WebResource.InitBuilder.class))).thenReturn(initDataObject);
+        final Response restResponse = apiTokenResource.revokeUserToken(request, response, limitedUser.getUserId());
+        Assert.assertNotNull(restResponse);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary use of fully qualified name 'Assert.assertNotNull' due to existing static import 'org.junit.Assert.assertNotNull'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5164094)